Framework/DataHandling/src/LoadILLIndirect2.cpp

```diff
@@ -273,6 +311,86 @@ void LoadILLIndirect2::loadDataIntoTheWorkSpace(NeXus::NXEntry &entry) {
   }
 }
 
+/**
+ * @brief LoadILLIndirect2::loadDiffractionData
+ * Load IN16B diffraction data from the Nexus file when requested.
+ * @param entry
+ */
+void LoadILLIndirect2::loadDiffractionData(NeXus::NXEntry &entry) {
+  NeXus::NXClass instrument = entry.openNXGroup("instrument");
+
+  // first, determine version
+  bool newVersion = instrument.containsDataSet("version");
+
+  // find the path to the data
+  std::string dataPath;
```

Comment:
These lines seem to be repeated, can you maybe create a function and call twice?